app/src/main/java/bg/zahov/app/ui/workout/start/StartWorkoutViewModel.kt

```diff
@@ -1,117 +1,229 @@
+            } else {
+                toastManager.showToast(R.string.starting_workout_while_another_is_active)
             }
         }
     }
 
-    fun startEmptyWorkout() {
+    /**
+     * Adds a duplicate template workout to the repository.
+     *
+     * This method creates a new `Workout` object based on the `StartWorkout` passed as input, appending the word "duplicate"
+     * to the name if the workout's name does not already contain the word "duplicate". The newly created workout is then
+     * added to the repository as a template workout.
+     *
+     * @param newWorkout The `StartWorkout` object that contains the details of the workout to duplicate.
+     */
+    fun addDuplicateTemplateWorkout(newWorkout: StartWorkout) {
         viewModelScope.launch {
-            workoutState.startWorkout(null)
+            repo.addTemplateWorkout(
+                Workout(
+                    id = generateRandomId(),
+                    name = if (newWorkout.name.contains("duplicate")) newWorkout.name else "${newWorkout.name} duplicate",
+                    duration = 0,
+                    volume = 0.0,
+                    date = LocalDateTime.now(),
+                    isTemplate = true,
+                    exercises = newWorkout.toWorkout(exerciseTemplates).exercises
+                )
+            )
         }
     }
 
-    fun startWorkoutFromTemplate(position: Int) {
+    /**
+     * Deletes a template workout from the repository.
+     *
+     * This method deletes the workout (converted from the `StartWorkout` object) from the repository, effectively removing
+     * the template workout.
+     *
+     * @param workout The `StartWorkout` object representing the workout to delete.
+     */
+    fun deleteTemplateWorkout(workout: StartWorkout) {
         viewModelScope.launch {
-            _templates.value?.get(position)?.let {
-                workoutState.startWorkout(it)
-            }
+            repo.deleteTemplateWorkout(workout.toWorkout(exerciseTemplates))
         }
     }
+}
 
-    fun deleteTemplateWorkout(position: Int) {
-        val list = _templates.value.orEmpty().toMutableList()
-        val toBeRemoved = list.removeAt(position)
-        viewModelScope.launch {
-            toBeRemoved.let { repo.deleteTemplateWorkout(it) }
-        }
-        _templates.value = list
+/**
+ * Converts a string representing a date and time into a `LocalDateTime` object.
+ *
+ * This function takes a string formatted as "HH:mm, d MMMM" (e.g., "13:00, 5 September") and parses it into a
+ * `LocalDateTime` object for the current year. This is useful when the input string does not include the year,
+ * but it is required for creating a `LocalDateTime` object.
+ *
+ * @return A `LocalDateTime` object that corresponds to the parsed date and time, with the current year.
+ */
+fun String.toLocalDateTime(): LocalDateTime =
+    LocalDateTime.parse(
+        "${LocalDate.now().year} $this",
```

Comment:
Looks like a hacky solution and as mentioned in the previous comment you don't really need it I believe.

Reply:
I cannot do the mapping from StartWorkout to Workout because of the way I implemented StartWorkout to hold the date as a plain string.

Reply:
Can you change it to LocalDateTime? Is this how it is on the back-end? If so, maybe keep those transformations in the Model layer and have the domain models use only the LocalDateTime.

Reply:
changed field to localDateTime in StartWorkout